File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        ArrayList<Character> arrlist = new ArrayList<>();

        for(int i = 0; i < s.length(); i++) {
            if(Character.isLetterOrDigit(s.charAt(i))) {
                arrlist.add(Character.toLowerCase(s.charAt(i)));
            }
        }

        int l = 0, r = arrlist.size() - 1;
        while(l < r) {
            if(!arrlist.get(l).equals(arrlist.get(r))) {
                return false;
            }
            l++;
            r--;
        }
        
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        letters = []

        for i in s:
            if i.isalnum():
                letters.append(i.lower())
        letters = "".join(letters)

        l, r = 0, len(letters) - 1
        while l <= r:
            if letters[l] != letters[r]:
                return False
            l += 1
            r -= 1
        return True